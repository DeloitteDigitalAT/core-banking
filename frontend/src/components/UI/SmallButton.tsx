import React from "react";

interface ButtonType {
  text: string;
  handler: () => void;
  active: boolean;
  extraClass?: string;
}

const SmallButton: React.FC<ButtonType> = properties => {
  const { text, handler, active, extraClass } = properties;
  return (
    <button
      type="button"
      className={`${
        active
          ? "bg-gradient-to-b from-green-100 to-green-200 border-green-100 text-xs hover:bg-green-400 hover:from-green-200 hover:to-green-300 border text-green-800"
          : "bg-gray-200 text-gray-400"
      } rounded-md px-2 py-0 transition duration-500 ease select-none focus:outline-none focus:shadow-outline ${extraClass}`}
      onClick={handler}
    >
      {text}
    </button>
  );
};

export default SmallButton;
