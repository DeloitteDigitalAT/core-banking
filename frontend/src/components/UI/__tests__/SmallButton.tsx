import React from "react";
import renderer from "react-test-renderer";
import SmallButton from "../SmallButton";

describe("UI/SmallButton component", () => {
  test("snapshot renders", () => {
    const tree = renderer.create(<SmallButton text="" handler={() => {}} active extraClass="" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
