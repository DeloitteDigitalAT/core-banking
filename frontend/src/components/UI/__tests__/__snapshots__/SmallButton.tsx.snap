// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/SmallButton component snapshot renders 1`] = `
<button
  className="bg-gradient-to-b from-green-100 to-green-200 border-green-100 text-xs hover:bg-green-400 hover:from-green-200 hover:to-green-300 border text-green-800 rounded-md px-2 py-0 transition duration-500 ease select-none focus:outline-none focus:shadow-outline "
  onClick={[Function]}
  type="button"
>
  
</button>
`;
